### PR TITLE
Add a sleep in Shellout.executeStream to reduce spinning.

### PR DESCRIPTION
**Pull request by rorygraves**
Whilst waiting for IO Shellout.executeStream was running a tight loop and consuming significant
amounts of CPU.  Adding a small sleep when no input was seen reduces cpu usage significant.

### CODE REVIEW

File: ops/src/main/scala/ammonite/ops/Shellout.scala [hunk 1]
```diff
@@ -65,14 +65,17 @@ object Shellout{
     val stdout = process.getInputStream
     val stderr = process.getErrorStream
     val chunks = collection.mutable.Buffer.empty[Either[Bytes, Bytes]]
+    val sources = Seq(stdout -> (Left(_: Bytes)), stderr -> (Right(_: Bytes)))
```
Comment (rorygraves) [inline ops/src/main/scala/ammonite/ops/Shellout.scala]: This prevents allocations during each cycle.

File: ops/src/main/scala/ammonite/ops/Shellout.scala [hunk 2]
```diff
@@ -81,6 +84,9 @@ object Shellout{
           ))
         }
       }
+      // if we did not read anything sleep briefly to avoid spinning
```
Comment (rorygraves) [inline ops/src/main/scala/ammonite/ops/Shellout.scala]: Only sleep if we have not seen input during this iteration.  This means that for fast processes we do not add unnecessary pauses.

**Comment by rorygraves** [comment]
@lihaoyi CI build failure looks unrelated

**Comment by lihaoyi** [comment]
Thanks!

**Pull request merged**